cms/djangoapps/contentstore/management/commands/tests/test_delete_course.py

```diff
@@ -60,3 +63,38 @@ def test_course_deleted(self):
             patched_yes_no.return_value = True
             call_command('delete_course', 'TestX/TS01/2015_Q1')
             self.assertIsNone(modulestore().get_course(SlashSeparatedCourseKey("TestX", "TS01", "2015_Q1")))
+
+    def test_course_deletion_with_keep_instructors(self):
+        """
+        Test --keep-instructors option,use of this option delete course from mongo store without
+        removing instructors.
+        """
+        course_id = SlashSeparatedCourseKey("TestX", "TS01", "2015_Q1")
+
+        staff_user=User.objects.create(
+            username = 'test_staff',
+            email = 'test_email@example.com',
+            is_staff = True,
+        )
+
+        instructor_user = User.objects.create(username='test_instructor', email='test_instructor@example.com')
+
+        # add instructor role
+        instructor_role = CourseInstructorRole(course_id)
+        auth.add_users(staff_user, instructor_role, instructor_user)
+
+        # Test instructor role is added successfully
+        self.assertGreater(len(instructor_role.users_with_role()), 0)
+
+        # Test if the course that is about to be deleted exists
+        self.assertIsNotNone(modulestore().get_course(course_id))
+
+        with mock.patch(self.YESNO_PATCH_LOCATION) as patched_yes_no:
+            patched_yes_no.return_value = True
```

Comment:
have you tried doing something like this => `with mock.patch(self.YESNO_PATCH_LOCATION, return_value=True) as patched_yes_no:`